### PR TITLE
Update ipython notebook examples

### PR DESCRIPTION
**Pull request by jdahm**
## Description

These were out of date since the removal of gt4py storages.

I also updated the backend names, since a few were out of date, and a few weren't present.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -36,7 +36,7 @@ IPython notebooks on Jupyter.
 
 .. code-block:: bash
 
-   $ pip install jupyterlab
+   $ pip install jupyterlab matplotlib
```
Comment (jdahm) [inline README.rst:39]: `matplotlib` is required for the ipython notebook examples.

**Comment by jdahm** [comment]
bors try

**Comment by havogt** [comment]
tests passed, just setting the status to green failed.

**Comment by gronerl** [comment]
bors try

**Review by gronerl — APPROVED** [review]

**Pull request merged**